CodeEdit/Features/CEWorkSpaceSettings/CEWorkspaceSettings/Models/CETask.swift

```diff
@@ -0,0 +1,141 @@
+//
+//  CETask.swift
+//  CodeEdit
+//
+//  Created by Axel Martinez on 2/4/24.
+//
+
+import Foundation
+
+/// CodeEdit task that will be executed by the task manager.
+class CETask: ObservableObject, Identifiable, Hashable, Codable {
```

Comment:
CETask is both a model and a viewmodel? Asking the question as it is in the models folder but also extends from ObservableObject and has Published variables

Reply:
It's used as a little of both due to the fact that it represents a single task *and* needs to update UI that needs to know about that task's information. I think the ergonomics would be weird to have a second struct/class representing a task, and using this to iterate through tasks and monitor task information should be okay.